lib/services/local/provisioning.go

```diff
@@ -71,9 +69,6 @@ func (s *ProvisioningService) tokenToItem(p types.ProvisionToken) (*backend.Item
 	if err := p.CheckAndSetDefaults(); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	if p.Expiry().IsZero() || p.Expiry().Sub(s.Clock().Now().UTC()) < time.Second {
-		p.SetExpiry(s.Clock().Now().UTC().Add(defaults.ProvisioningTokenTTL))
-	}
```

Comment:
This will be a breaking change for any third party clients of the `UpsertToken` api relying on the default expiry. `UpsertToken` is is a part of `api/client` and token creation seems likely to have been automated by some users. It would be unfortunate to have your ephemeral token script suddenly start generating tokens that stick around forever.

We could ship this in teleport 13 with a release note, or add a new `expiryless` (or similar) flag to `ProvisionToken` so that the default behaviour doesn't change, or accept this as a breaking change and move on. Not sure where I stand on that but I think it's worth making a conscious decision.

Reply:
That's true - I hadn't thought RE: external automations people had configured.

My gut feeling is to ship this in a major release and then just announce it as loud as we can as part of upgrade notes.

I think adding an addition field to ProvisionToken feels like it makes it more unwieldy, and just adds additional logic across the codebase we need to account for when displaying these values and when processing tokens on join.